test/specs/box.spec.js

```diff
@@ -1,14 +1,93 @@
 describe('Box annotation', function() {
   describe('auto', jasmine.fixtures('box'));
 
+  const eventIn = function(xScale, yScale, element) {
+    const options = element.options;
+    const adjust = options.borderWidth / 2 - 1;
+    return {x: element.x, y: element.y - adjust};
+  };
+  const eventOut = function(xScale, yScale, element) {
+    const options = element.options;
+    const adjust = options.borderWidth / 2 + 1;
+    return {x: element.x, y: element.y - adjust};
+  };
+
   window.testEvents({
     type: 'box',
     id: 'test',
     xMin: 2,
     yMin: 2,
     xMax: 4,
     yMax: 4,
-    borderWidth: 0
-  });
+    borderWidth: 10
+  }, eventIn, eventOut);
+
+  describe('events, removing borderWidth by callback, ', function() {
```

Comment:
@stockiNail what are these use cases, and why should we support these?

Reply:
Someone can remove the option from `element.options` by the event callback.
This is it has been suggested by #589, where the user can change the element options therefore er are not sure that the borderWidth can have a consistent value. In fact, this is tested and use 0 if borderWidth / 2 is NaN

Reply:
This is also true for other options which can be changed at runtime without passing thru the `update` and therefore the code can be broken for that.

Reply:
User can delete everything in the element, but we need to consider supported use cases vs stupid users :)

Reply:
for example, I'd not support something like `delete element.inRange`. And deleting borderWidth goes in same category. It'd be ok to set it to a new value, like `0`, but why delete?

Reply:
> for example, I'd not support something like `delete element.inRange`. And deleting borderWidth goes in same category. It'd be ok to set it to a new value, like `0`, but why delete?

If someone is not familiar with fallback, it could think that remove it the default will be used.

Reply:
> User can delete everything in the element, but we need to consider supported use cases vs stupid users :)

I agree but the users are users... that's my opinion. Usually I try to do something doesn't get broken, whoever is the user and use case. In the other project, this approach reduces the amount of issues (usually).

Reply:
@kurkle anyway I think we can remove the test cases and also the check in the code. Let me know.